pkg/controllers/tlspolicy/tlspolicy_controller.go

```diff
@@ -220,6 +246,33 @@ func (r *TLSPolicyReconciler) readyCondition(targetNetworkObjectectKind string,
 	return cond
 }
 
+func (r *TLSPolicyReconciler) updateGatewayCondition(ctx context.Context, condition metav1.Condition, gatewayDiff *reconcilers.GatewayDiff) error {
+
+	// update condition if needed
+	for _, gw := range append(gatewayDiff.GatewaysWithValidPolicyRef, gatewayDiff.GatewaysMissingPolicyRef...) {
+		previous := gw.DeepCopy()
+		meta.SetStatusCondition(&gw.Status.Conditions, condition)
+		if !reflect.DeepEqual(previous.Status.Conditions, gw.Status.Conditions) {
+			if err := r.Client().Status().Update(ctx, gw.Gateway); err != nil {
+				return err
+			}
+		}
+	}
+
+	// remove condition from gateway that is no longer referenced
+	for _, gw := range gatewayDiff.GatewaysWithInvalidPolicyRef {
+		previous := gw.DeepCopy()
+		meta.RemoveStatusCondition(&gw.Status.Conditions, condition.Type)
+		if !reflect.DeepEqual(previous.Status.Conditions, gw.Status.Conditions) {
+			if err := r.Client().Status().Update(ctx, gw.Gateway); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
```

Comment:
This is duplicated from the DNSPolicy controller. I can look at re-factoring this now, but it's probably preferable to keep these changes to a minimum. I'd also be interested in finding out if the kuadrant data plane team have added this to their controllers/policies, or plan to, since we should ideally just all be doing this in the same way.